pkg/virt-launcher/virtwrap/network/podnic.go

```diff
@@ -194,12 +198,20 @@ func (l *podNIC) PlugPhase1() error {
 	if dhcpConfig != nil {
 		log.Log.V(4).Infof("The generated dhcpConfig: %s", dhcpConfig.String())
 		if err := l.cacheFactory.CacheDHCPConfigForPid(getPIDString(l.launcherPID)).Write(l.podInterfaceName, dhcpConfig); err != nil {
-			log.Log.Reason(err).Error("failed to save dhcpConfig configuration")
-			return errors.CreateCriticalNetworkError(err)
+			return fmt.Errorf("failed to save DHCP configuration: %w", err)
 		}
 	}
-
 	domainIface := l.infraConfigurator.GenerateDomainIfaceSpec()
+
+	if err := l.storeCachedDomainIface(domainIface); err != nil {
```

Comment:
I would store only if `domainIface` is not `nil`. Similar to `dhcpConfig`.

Reply:
It returns a struct not a pointer to struct

```golang
GenerateDomainIfaceSpec() api.Interface
```

Reply:
talking to myself: https://github.com/kubevirt/kubevirt/pull/5943#discussion_r673913683